File: src/main/webapp/app/module/secondary/RestPreset.ts
import { Preset } from '@/module/domain/Preset';
import { RestPresetModuleToApply, mapPresetModulesToApply } from '@/module/secondary/RestPresetModuleToApply';

export interface RestPreset {
  name: string;
  modules: RestPresetModuleToApply[];
}

export const mapToPreset = (restPreset: RestPreset): Preset => ({
  name: restPreset.name,
  modules: mapPresetModulesToApply(restPreset.modules),
});

File: src/main/java/tech/jhipster/lite/module/infrastructure/primary/RestPreset.java
package tech.jhipster.lite.module.infrastructure.primary;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.Collection;
import tech.jhipster.lite.module.domain.preset.Preset;

@Schema(name = "Preset", description = "Information on specific configuration")
record RestPreset(
  @Schema(description = "Name of specific configuration") String name,
  @Schema(description = "Modules to apply of this specific configuration") Collection<RestModuleToApply> modules
) {
  public static RestPreset from(Preset preset) {
    return new RestPreset(preset.name().name(), preset.modules().modules().stream().map(RestModuleToApply::from).toList());
  }
}
